src/common/com/intellij/plugins/haxe/lang/util/HaxePsiUtil.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2018 Ilya Malanin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intellij.plugins.haxe.lang.util;
+
+import com.intellij.lang.ASTNode;
+import com.intellij.plugins.haxe.lang.lexer.HaxeTokenTypeSets;
+import com.intellij.plugins.haxe.lang.lexer.HaxeTokenTypes;
+import com.intellij.plugins.haxe.lang.psi.*;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.tree.IElementType;
+import com.intellij.psi.util.PsiTreeUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class HaxePsiUtil {
+  @Nullable
+  public static IElementType getOperationType(@NotNull HaxeExpression expression) {
+    final ASTNode token = expression.getNode().findChildByType(HaxeTokenTypeSets.UNARY_OPERATORS);
+    if (token != null) return token.getElementType();
+    return null;
+  }
+
+  public static boolean isIncrementDecrementOperation(@Nullable PsiElement element) {
+    if (element instanceof HaxePrefixExpression) {
+      final IElementType operationType = HaxePsiUtil.getOperationType((HaxePrefixExpression)element);
+      return HaxeTokenTypeSets.UNARY_READ_WRITE_OPERATORS.contains(operationType);
+    }
+    else return element instanceof HaxeSuffixExpression;
+  }
+
+  @Nullable
+  public static PsiElement skipParenthesizedExprUp(@Nullable PsiElement parent) {
```

Comment:
This is the only function that is PSI-specific, and it can be private because it's only used in this class.  Otherwise, you might actually want it in UsefulPsiTreeUtil.